components/admin/Usertable.js

```diff
@@ -0,0 +1,283 @@
+import React from "react";
+import MaterialTable from "material-table";
+import { Button, Grid, Typography, Paper } from "@material-ui/core";
+import TextField from "@material-ui/core/TextField";
+import Dialog from "@material-ui/core/Dialog";
+import DialogActions from "@material-ui/core/DialogActions";
+import DialogContent from "@material-ui/core/DialogContent";
+import DialogContentText from "@material-ui/core/DialogContentText";
+import DialogTitle from "@material-ui/core/DialogTitle";
+import MenuItem from "@material-ui/core/MenuItem";
+import FormControl from "@material-ui/core/FormControl";
+import Select from "@material-ui/core/Select";
+import InputLabel from "@material-ui/core/InputLabel";
+import UserData from "./UserData.json";
+
+const style = { Paper: { padding: 20, margin: 20 } };
+
+export default function Usertable() {
+  const [state, setState] = React.useState({
+    inviteUserOpen: false,
+    addUserOpen: false,
+    columns: [
+      { title: "First Name", field: "firstName" },
+      { title: "Last Name", field: "lastName" },
+      { title: "Email", field: "email" },
+      {
+        title: "User Type",
+        field: "userType",
+        lookup: { 0: "Admin", 1: "Member" }
+      }
+    ],
+    data: UserData
+  });
+
+  const [values, setValues] = React.useState({
+    inputFirstName: "",
+    inputLastName: "",
+    inputEmail: "",
+    inputPassword: "",
+    userType: ""
+  });
+
+  const handleChange = event => {
+    setValues(oldValues => ({
+      ...oldValues,
+      [event.target.name]: event.target.value
+    }));
+  };
+
+  function handleInviteClick() {
```

Comment:
Since I'm using hooks in this case, I cannot simply call setStates({propertyName: new_property}); this will replace the entire state with {propertyName: new_property}

Reply:
Gotchu!